### PR TITLE
chore(build): optimize the type conversion of tag plugin

### PR DESCRIPTION
**Pull request by tolking**
Link #11580

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by tolking** [comment]
The `helper.ts` file is use to generate code prompt files.

After we add the `tag` plugin to the document, we need to remove the side effects brought by the `tag` plugin here.

File: internal/build/src/tasks/helper.ts [hunk 1]
```diff
@@ -27,18 +27,25 @@ const typeMap = {
   vue: ['Component', 'VNode', 'CSSProperties', 'StyleValue'],
 }
 
-const reComponentName: ReComponentName = (title) =>
-  `el-${hyphenate(title).replace(/[ ]+/g, '-')}`
+const removeTag = (str: string) => {
+  return str.replaceAll(/\^\([^)]*\)/g, '').trim()
```
Comment (chenxch) [inline internal/build/src/tasks/helper.ts:31]: Are `replaceAll` and `g` duplicates?

Comment (btea) [reply internal/build/src/tasks/helper.ts:31]: That's right, you can refer to [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll#:~:text=If%20pattern%20is%20a%20regex).

**Review by btea — APPROVED** [review]

**Pull request merged**